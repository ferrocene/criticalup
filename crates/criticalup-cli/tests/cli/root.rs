// SPDX-FileCopyrightText: The Ferrocene Developers
// SPDX-License-Identifier: MIT OR Apache-2.0

use crate::assert_output;
use crate::utils::TestEnvironment;

#[tokio::test]
async fn no_args() {
    let test_env = TestEnvironment::prepare().await;
    assert_output!(test_env.cmd());
}

#[tokio::test]
async fn help_flags() {
    let test_env = TestEnvironment::prepare().await;

    let no_args = test_env.cmd().output().await.unwrap();
    let help_short = test_env.cmd().arg("-h").output().await.unwrap();
    let help_long = test_env.cmd().arg("--help").output().await.unwrap();

    assert_eq!(&no_args.stdout, &help_short.stdout);
    assert_eq!(&no_args.stdout, &help_long.stdout);

    assert_eq!(&no_args.stderr, &help_short.stderr);
    assert_eq!(&no_args.stderr, &help_long.stderr);

    assert!(help_short.status.success());
    assert!(help_long.status.success());
}

#[tokio::test]
async fn version_flags() {
    let test_env = TestEnvironment::prepare().await;

    let version_short = test_env.cmd().arg("-V").output().await.unwrap();
    let version_long = test_env.cmd().arg("--version").output().await.unwrap();
    assert_eq!(version_long, version_short);

    assert_output!(test_env.cmd().arg("--version"));
}
