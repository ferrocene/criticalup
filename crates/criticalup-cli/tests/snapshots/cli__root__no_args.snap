---
source: crates/criticalup-cli/tests/cli/root.rs
expression: repr
---
exit: exit status: 1

empty stdout

stderr
------
CriticalUp is the official tool to download and install Ferrocene

Usage:
  criticalup-test [OPTIONS] <COMMAND>

Commands:
  auth     Show and change authentication with the download server
  install  Install the toolchain for the given project based on the manifest `criticalup.toml`
  clean    Delete cache and unused installations
  run      Run a command for a given toolchain
  remove   Delete all the products specified in the manifest `criticalup.toml`
  verify   Verify a given toolchain
  which    Display which binary will be run for a given command
  archive  Create a tar archive of the toolchain based on the manifest `criticalup.toml`
  doc      Open the documentation for the current toolchain

Options:
  -v, --verbose...                  Enable debug logs, -vv for trace
      --log-level [<LOG_LEVEL>...]  Tracing directives
  -h, --help                        Print help
  -V, --version                     Print version
------
